src/registry.cpp

```diff
@@ -61,6 +61,8 @@ std::string Registry::login(const std::string &username,
         UserProfile::get()->getAuthManager()->saveCredentials(
             url, AuthCredentials(username, password));
 
+        this->authToken = token;
```

Comment:
Ok, be careful though.. long sessions (think long uploads from a rural area with dial-up speeds) could fail if the token expires. Login has now side effects.

Reply:
It was probably a problem before too, but now you have a state variable.

Reply:
A good check in `getAuthToken` would be to see if the token has expired, and if so, to re-do the login and return a refreshed token (now that we have the `expires` field...)

Reply:
Done!